### PR TITLE
Make StepNode just extend higher-level API in workflow-api

### PR DESCRIPTION
**Pull request by svanoort**
Allows compatibility while permitting workflow-api to expose step info. 

Downstream of https://github.com/jenkinsci/workflow-api-plugin/pull/28

@reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -75,7 +75,8 @@
         <dependency>
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-api</artifactId>
-            <version>2.6</version>
+            <!-- Temporary version to pull in upstream PRs, will revert before merge once that is released -->
+            <version>2.9-stepnode-SNAPSHOT</version>
```
Comment (jglick) [inline pom.xml]: I find it helpful to link to the upstream PR in a comment.

File: src/main/java/org/jenkinsci/plugins/workflow/cps/nodes/StepNode.java [hunk 2]
```diff
@@ -7,16 +7,9 @@
 import javax.annotation.CheckForNull;
 
 /**
- * Optional interface for {@link FlowNode} that has associated {@link StepDescriptor}
- *
- * @author Kohsuke Kawaguchi
+ * Optional interface for {@link FlowNode} that has associated {@link StepDescriptor}.
+ * Retained for binary & API compatibility but the interface has been pushed up a level into the
+ *  workflow-api plugin to allow broader access.
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/cps/nodes/StepNode.java]: Just mark it `@Deprecated`.

File: src/main/java/org/jenkinsci/plugins/workflow/cps/nodes/StepNode.java [hunk 3]
```diff
@@ -1,22 +1,18 @@
...
  */
-public interface StepNode {
-    /**
-     * Returns the descriptor for {@link Step} that produced this flow node.
-     *
-     * @return
-     *      null for example if the descriptor that created the node has since been uninstalled.
-     */
-    @CheckForNull StepDescriptor getDescriptor();
+@Deprecated
```
Comment (christ66) [inline src/main/java/org/jenkinsci/plugins/workflow/cps/nodes/StepNode.java]: :NIT: would be nice to java javadoc suggesting to use the workflow-api interface instead.

**Comment by christ66** [comment]
:bee:

**Comment by christ66** [comment]
:bee:

**Pull request merged**